src/binary/decimal.rs

```diff
@@ -69,9 +72,9 @@ where
             }
         } else {
             // Otherwise, allocate a Vec<u8> with the necessary representation.
-            let mut coefficient_bytes = match decimal.coefficient.magnitude() {
-                UInt::U64(unsigned) => unsigned.to_be_bytes().into(),
-                UInt::BigUInt(big) => big.to_bytes_be(),
+            let mut coefficient_bytes = match &decimal.coefficient.magnitude().data {
+                UIntData::U64(unsigned) => unsigned.to_be_bytes().into(),
+                UIntData::BigUInt(big) => big.to_bytes_be(),
```

Comment:
🗺️ This diff contains several instances of this change. Instead of matching directly on the `UInt`, we now match on its private inner `UIntData` field.